### PR TITLE
fix: prevent blocked addresses from sending ICS 20 transfers

### PR DESCRIPTION
**Pull request by colin-axner**
## Description

Prevent IBC transfers from being sent by a blocked module account address. If the module account address is blocked it would not be capable of receiving a refund and thus should not be allowed to send IBC transfers. Module accounts which would like to interfact with ICS 20 should ensure they do not need to be blocked

closes: #XXXX

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [x] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/ibc-go/blob/master/CONTRIBUTING.md#pr-targeting))
- [x] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [x] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/master/docs/building-modules/structure.md).
- [x] Wrote unit and integration [tests](https://github.com/cosmos/ibc-go/blob/master/CONTRIBUTING.md#testing)
- [x] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`)
- [x] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [x] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [x] Re-reviewed `Files changed` in the Github PR explorer
- [x] Review `Codecov Report` in the comment section below once CI passes

### CODE REVIEW

File: modules/apps/transfer/keeper/relay.go [hunk 1]
```diff
@@ -48,6 +48,8 @@ import (
 // 4. A -> C : sender chain is sink zone. Denom upon receiving: 'C/B/denom'
 // 5. C -> B : sender chain is sink zone. Denom upon receiving: 'B/denom'
 // 6. B -> A : sender chain is sink zone. Denom upon receiving: 'denom'
+//
+// Note: An IBC Transfer must be initiated using a MsgTransfer via the Transfer rpc handler
 func (k Keeper) SendTransfer(
```
Comment (colin-axner) [inline modules/apps/transfer/keeper/relay.go:53]: I will do a followup pr to make this private for the next major release

**Review by damiannolan — APPROVED** [review]

File: modules/apps/transfer/keeper/relay.go [hunk 2]
```diff
@@ -62,6 +64,10 @@ func (k Keeper) SendTransfer(
 		return types.ErrSendDisabled
```
Comment (crodriguezvega) [inline modules/apps/transfer/keeper/relay.go:64]: This is a nit and no need to block the PR on this, but now that you added tests for the msg server, maybe we can also had a test case for the situation where send is disabled.

**Review by crodriguezvega — APPROVED** [review]
Thank you, @colin-axner!

File: modules/apps/transfer/keeper/relay.go (hunk 2 above)
Comment (colin-axner) [reply modules/apps/transfer/keeper/relay.go:64]: looks like this test case is missing for receive as well. Would prefer it be done in a separate pr

**Review by chatton — APPROVED** [review]
LGTM

**Review by seantking — APPROVED** [review]

**Pull request merged**